magento/magento2-page-builder
# magento/magento2-page-builder#842: Template Preview Images Incorrectly Saved to Media Directory

### Description (*)

This fixes an issue where Page Builder would save its preview images into the top-level `media` directory, as opposed to within its designated `.template-manager` subdirectory. It also contains some minor refactoring.

### Story

_N/A_

### Bug

_N/A_

### Task

_N/A_

### Fixed Issues

1. magento/magento2-page-builder#842: Template Preview Images Incorrectly Saved to Media Directory

### Builds

_N/A_

### Related Pull Requests

_N/A_

### Manual Testing Scenarios (*)

- [ ] Validate Saved to Directory

1. Click on _any_ Page Builder-enabled field (doesn't have to be empty)
2. Click **Save as Template**
3. Define a Template Name (such as "Example")
4. Click **Save**
5. Verify that "The current contents of Page Builder has been successfully saved as a template."
6. Click **Apply Template**
7. Inspect the Preview Image path (can be full-size _or_ thumbnail) of the template you just saved
8. Validate that the image was saved _within_ the `.template-manager` directory

- [ ] Validate Deleted from Directory

1. Follow Steps 1-8 from the "Validate Saved to Directory" scenario
2. Browse to **Content** » **Templates**
3. Open the Preview Image in a new tab ("Open Image in New Tab")
4. Click **Delete**
5. Verify that you see, "Are you sure you want to permanently delete template Example" (_where "Example" is your Template Name_)
6. Click **OK**
7. Verify that you see, "Template successfully deleted."
8. Reload the tab from Step 3 pointing to the Preview Image
9. Validate that it now returns a 404

### Questions or Comments

Other minor improvements that can be made here include:

- amending the messaging from "The current contents of Page Builder **has** been successfully saved as a template." to "The current contents of Page Builder **have** been successfully saved as a template."
- reducing the complexity of the `storePreviewImage` method by creating helper methods which handle image naming conventions, etc.
- improving instances of `str_replace('.jpg', '-thumb.jpg', ...)` so that other file formats (such as PNG) can be used, particularly via data patches.

### Checklist
